### PR TITLE
Use dev crayon and cli

### PR DESCRIPTION
**Pull request by jennybc**
Related to issue #1296

### CODE REVIEW

**Comment by jennybc** [comment]
I'm not clear whether it was ever intentional to save snapshots with ANSI escapes (linked below and, in fact, I'm having the *opposite* problem in reprex):

* [tests/testthat/_snaps/expect-constant.md](https://github.com/r-lib/testthat/blob/55a14ab7cb580749d1bc6b537203fc7dfab91124/tests/testthat/_snaps/expect-constant.md)
* [tests/testthat/_snaps/expect-equality.md](https://github.com/r-lib/testthat/blob/55a14ab7cb580749d1bc6b537203fc7dfab91124/tests/testthat/_snaps/expect-equality.md)

**Comment by gaborcsardi** [comment]
FWIW dev cli is cleaner and it does not depend on crayon any more. So if we are switching things around already, we might just start using that.

File: tests/testthat/_snaps/snapshot-serialize.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# snapshot output doesn't use ANSI escapes
+
+    Code
+      cli::cli_alert_success("Success!")
+    Message <cliMessage>
+      v Success!
+
+---
+
```
Comment (jennybc) [inline tests/testthat/_snaps/snapshot-serialize.md:9]: Why does `expect_snapshot_output()` produce nothing here?

Comment (hadley) [reply tests/testthat/_snaps/snapshot-serialize.md:9]: It's a message not output? Perhaps `expect_snapshot_output()` is confusingly named?

**Comment by hadley** [comment]
My goal was to turn ANSI escapes should be off by default, but allow you to turn them on if you want them with `local_reproducible_output()`

File: tests/testthat/_snaps/snapshot-serialize.md (hunk 1 above)
Comment (jennybc) [reply tests/testthat/_snaps/snapshot-serialize.md:9]: I guess I had different expectations based on this in the docs:

> `expect_snapshot_output()` captures just output printed to the console

**Comment by hadley** [comment]
I think this is now resolved via other PRs?

**Comment by jennybc** [comment]
Yeah this was part of the process to figure out what was going on cli and snapshot tests. Fine to close and delete the branch.

**Pull request closed without merging**